src/read.c

```diff
@@ -4357,7 +4353,26 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
         }
     }
     assert(alphaItemCount == colorItemCount);
-    const avifResult result = avifMetaFindOrCreateItem(meta, maxItemID + 1, alphaItem); // Create new empty item.
+    // Find an unused ID.
+    uint32_t newItemID = 0;
+    avifBool isUsed;
+    do {
+        ++newItemID;
+        isUsed = AVIF_FALSE;
+        for (uint32_t i = 0; i < meta->items.count; ++i) {
+            if (meta->items.item[i]->id == newItemID) {
+                isUsed = AVIF_TRUE;
+                break;
+            }
+        }
+    } while (isUsed && newItemID != 0);
+    avifResult result;
+    if (newItemID == 0) {
+        // In the improbable case where all IDs are used.
```

Comment:
Some test such as the following might be clearer than implicit uint32_t looping overflow:

```c
AVIF_CHECKERR(meta->items.count < MAX_UINT32_T - 1, AVIF_RESULT_DECODE_ALPHA_FAILED);
```

Reply:
Done